### PR TITLE
Make templates more customizable

### PR DESCRIPTION
**Pull request by enyst**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This PR proposes to make the Jinja2 templates more customizable.
- it breaks down examples.j2 from user_prompt.j2, to allow a different user prompt if wanted without affecting the examples and the other way around
- it breaks down commands a bit more
- agentskills refactoring of the way they're rendered
- placeholders for tool use

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by enyst** [comment]
@xingyaoww I'm thinking that this is just a step. We need some more splitting IMHO - I restructured a bit more, in blocks too, but I didn't get to finish/test those.

In addition to these user and default directories, how do you think about one .md/.j2 file per _block_ (e.g. load `EXECUTE_BASH` from user-directory if present, otherwise default, etc for all blocks we can figure out), would it be too granular?

**Review by xingyaoww — COMMENTED** [review]
Yep - I very much like the idea of making them into different file then assemble them together. How about we create a folder `prompts` to put these?

**Comment by ngduyanhece** [comment]
**Overall Feedback:**

The PR introduces significant improvements to the customization of Jinja2 templates, enhancing flexibility and modularity. The changes are well-structured, allowing for more granular control over template components. However, there are areas where error handling and code clarity can be improved, particularly in the new functions and template rendering logic.

**Score:** 85/100

**Labels:** Enhancement, Tests

**Code Suggestions:**

1. **File:** `openhands/agenthub/codeact_agent/agent_skills.j2`
   - **Language:** Jinja
   - **Suggestion Content:** Add error handling for missing skill documentation.
   - **Relevant Line:** `+{{ get_skill_docstring(skill_name) }}`
   - **Existing Code:**
     ```jinja
     {{ get_skill_docstring(skill_name) }}
     ```
   - **Improved Code:**
     ```jinja
     {% set docstring = get_skill_docstring(skill_name) %}
     {{ docstring if docstring else 'Documentation not available' }}
     ```

2. **File:** `openhands/agenthub/codeact_agent/codeact_agent.py`
   - **Language:** Python
   - **Suggestion Content:** Improve error handling in `_get_skill_docstring` to log errors instead of printing them.
   - **Relevant Line:** `+            print(e)`
   - **Existing Code:**
     ```python
     print(e)
     ```
   - **Improved Code:**
     ```python
     logging.error(f"Error retrieving skill docstring for {skill_name}: {e}")
     ```

3. **File:** `tests/unit/test_prompt_manager.py`
   - **Language:** Python
   - **Suggestion Content:** Ensure that the test for loading agent skills checks for all expected skills.
   - **Relevant Line:** `+    assert ('open_file(path: str, line_number: int | None = 1, context_lines: int | None = 100) -> None' in manager.system_message)`
   - **Existing Code:**
     ```python
     assert ('open_file(path: str, line_number: int | None = 1, context_lines: int | None = 100) -> None' in manager.system_message)
     ```
   - **Improved Code:**
     ```python
     expected_skills = [
         'open_file(path: str, line_number: int | None = 1, context_lines: int | None = 100) -> None',
         # Add other expected skills here
     ]
     for skill in expected_skills:
         assert skill in manager.system_message
     ```

The PR enhances the customization of Jinja2 templates effectively. Here are some suggestions:

1. In `agent_skills.j2`, consider adding error handling for missing skill documentation.
2. Improve error handling in `_get_skill_docstring` by logging errors instead of printing them.
3. Ensure that the test for loading agent skills checks for all expected skills.

File: openhands/agenthub/codeact_agent/agent_skills.j2 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{% for skill_name in available_skills %}
+{{ get_skill_docstring(skill_name) }}
```
Comment (ngduyanhece) [inline openhands/agenthub/codeact_agent/agent_skills.j2:2]: **Suggestion:** Add error handling for missing skill documentation.

**Existing Code:**
```Jinja
{{ get_skill_docstring(skill_name) }}
```

**Improved Code:**
```Jinja
{% set docstring = get_skill_docstring(skill_name) %}
{{ docstring if docstring else 'Documentation not available' }}
```

**Details:**
Add error handling for missing skill documentation.

**Pull request closed without merging**